### ISSUE TITLE
java.lang.IllegalStateException: Exception occurred while trying to download a stub for group [com.example] module [contract-rest-service] and classifier [stubs] in []

### ISSUE THREAD

**Issue by belon** [spring-guides/gs-contract-rest#4]
Complete example doesn't work using gradle.

When I try to run command `./gradlew clean build` in complete directory there is an error related with finding stub generated by `contract-rest-service` module.

```
# ./gradlew clean build

> Task :contract-rest-client:test

hello.ContractRestClientApplicationTest > get_person_from_service_contract FAILED
    java.lang.IllegalStateException
        Caused by: org.eclipse.aether.resolution.ArtifactResolutionException
            Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException
```

**Comment by belon** [comment]
@dsyer thanks for your fix. This solve the issue with installing stub locally and I am able to use it in test.

I tried to upload the stub to our nexus server and read it from remote server. I set `workOffline(false)` in file `ContractRestClientApplicationTest` but then test failed with exception
```
java.lang.IllegalStateException: No stubs were found on classpath for [com.example:contract-rest-service]
```
I also added dependency `testCompile('com.example:contract-rest-service:0.0.2-RELEASE')` to `contract-rest-client/build.gradle` file but this doesn't help.

Do you know if it is possible to use stub without installing it manually in local maven?

**Comment by dsyer** [comment]
> Do you know if it is possible to use stub without installing it manually in local maven?

Yes, it is, and I'm pretty sure you can read the user guide to find the details. Isn't the version 0.0.1-SNAPSHOT?

**Comment by belon** [comment]
Yes version is `0.0.1-SNAPSHOT`. I think that I found the issue.
There is no option to overwrite `username` and `password` for remote repository using `StubRunnerRule` class. There is method to set `options` but this method will not copy all the options only some of them.

Example usage:
```
private StubRunnerOptions options = new StubRunnerOptionsBuilder()
			.withPassword("pass")
			.withUsername("username")
			.withPort(8100)
			.withStubRepositoryRoot("http://url-to-nexus.com/content/repositories/public")
			.withWorkOffline(false)
			.build();

	@Rule
	public StubRunnerRule stubRunnerRule = new StubRunnerRule()
		.options(options);
```

Method `options()` looks like that:
```
@Override public StubRunnerRule options(StubRunnerOptions stubRunnerOptions) {
		builder().withOptions(stubRunnerOptions);
		return this.delegate;
	}
```
and method `withOptions` looks like that
```
public StubRunnerOptionsBuilder withOptions(StubRunnerOptions options) {
		this.minPortValue = options.minPortValue;
		this.maxPortValue = options.maxPortValue;
		this.stubRepositoryRoot = options.stubRepositoryRoot;
		this.workOffline = options.workOffline;
		this.stubsClassifier = options.stubsClassifier;
		return this;
	}
```

Only some of the fields are copied from passed options parameter. This is a little bit confused.
I see that by default values for `username` and `password` are taken from system properties. Would be good to have an option to overwrite these values if needed.

**Comment by dsyer** [comment]
This is now way off topic for the guide right? I'm still interested in fixing the guide so `gradle build` works out of the box if anyone has any suggestions.

**Comment by belon** [comment]
Yes you have right. I will create a ticket in spring-cloud-contract project. Anyway thanks for your help.
